# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/nsDragServiceProxy.h
# Commit: 384b025f56d3
# Full Hash: 384b025f56d3f87e045ecede6b1a16b71c0b5d4f
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 21 - Separate nsIDragService and nsIDragSession implementations r=gstoll,geckoview-reviewers,rkraesig,win-reviewers,m_kato
#   
#   Split the class inheritance trees for nsIDragService and nsIDragSession.
#   Remember that, before the start of this patch series, the inheritance
#   diagram was:
# ==============================================================================

diff -r beeda5ef3712 -r 384b025f56d3 widget/nsDragServiceProxy.h
--- a/widget/nsDragServiceProxy.h	Thu Jul 04 07:48:11 2024 +0000
+++ b/widget/nsDragServiceProxy.h	Thu Jul 04 07:48:11 2024 +0000
@@ -8,26 +8,48 @@
 
 #include "nsBaseDragService.h"
 
-// Temporary inheritance from nsBaseDragService instead of nsBaseDragSession
-// (which nsBaseDragService temporarily inherits).
-// This will be undone at the end of this patch series.
-class nsDragSessionProxy : public nsBaseDragService {};
+class nsDragSessionProxy : public nsBaseDragSession {
+ public:
+  NS_INLINE_DECL_REFCOUNTING_INHERITED(nsDragSessionProxy, nsBaseDragSession)
 
-// Temporary inheritance from nsDragSessionProxy instead of nsBaseDragService
-// (which nsDragSession temporarily inherits).
-// This will be undone at the end of this patch series.
-class nsDragServiceProxy final : public nsDragSessionProxy {
- public:
-  nsDragServiceProxy();
+  MOZ_CAN_RUN_SCRIPT virtual nsresult InvokeDragSession(
+      nsIWidget* aWidget, nsINode* aDOMNode, nsIPrincipal* aPrincipal,
+      nsIContentSecurityPolicy* aCsp, nsICookieJarSettings* aCookieJarSettings,
+      nsIArray* aTransferableArray, uint32_t aActionType,
+      nsContentPolicyType aContentPolicyType) override;
 
-  NS_INLINE_DECL_REFCOUNTING_INHERITED(nsDragServiceProxy, nsDragSessionProxy)
-
-  // nsBaseDragService
-  virtual nsresult InvokeDragSessionImpl(
+  nsresult InvokeDragSessionImpl(
       nsIWidget* aWidget, nsIArray* anArrayTransferables,
       const mozilla::Maybe<mozilla::CSSIntRegion>& aRegion,
       uint32_t aActionType) override;
 
+  void SetDragTarget(mozilla::dom::BrowserChild* aTarget);
+
+  MOZ_CAN_RUN_SCRIPT
+  nsresult EndDragSessionImpl(bool aDoneDrag, uint32_t aKeyModifiers) override;
+
+ private:
+  ~nsDragSessionProxy();
+
+  // The source for this drag.  This is null if the source is a different
+  // application or drag session.
+  nsWeakPtr mSourceBrowser;
+  // The target for this drag.  This is null if the target is a different
+  // application or drag session.
+  nsWeakPtr mTargetBrowser;
+};
+
+class nsDragServiceProxy : public nsBaseDragService {
+ public:
+  NS_INLINE_DECL_REFCOUNTING_INHERITED(nsDragServiceProxy, nsBaseDragService)
+
+  already_AddRefed<nsIDragSession> CreateDragSession() override;
+
+  NS_IMETHOD StartDragSession(nsISupports* aWidgetProvider) override;
+
+  NS_IMETHOD GetCurrentSession(nsISupports* aWidgetProvider,
+                               nsIDragSession** aSession) override;
+
  private:
   virtual ~nsDragServiceProxy();
 };